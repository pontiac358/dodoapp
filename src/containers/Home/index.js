import React from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import { fetchAllProducts } from '../../actions/products'
import ProductList from '../Products/components/ProductList'
import styled from '@emotion/styled'

const Container = styled.div`
    max-width:1100px;
    margin:auto;   
`;

const Title = styled.h2`
    font-size: 26px;
    text-align:center;
    margin-bottom: 50px;  
`;
Title.displayName = 'Title';

export class Home extends React.Component {
    componentDidMount(){
        this.props.fetchAllProducts()
    }

    render() {
        const { productsAll } = this.props;
        return (
            <Container>
                {
                    productsAll.data && productsAll.data.map((products, key) => {
                        return(
                            <React.Fragment key={ key }>
                                <Title>{ products.title }</Title>
                                <ProductList products={ products.data } />
                            </React.Fragment>
                        )
                    })
                }
            </Container>
        );
    }
}

Home.propTypes = {
    productsAll: PropTypes.shape({
        data: PropTypes.array,
        isLoading: PropTypes.bool
    }),
};

const mapStateToProps = (state) => {
    return {
        productsAll: state.productsAll,
    }
};

const mapDispatchToProps = {
    fetchAllProducts
};

export default connect(mapStateToProps, mapDispatchToProps)(Home)