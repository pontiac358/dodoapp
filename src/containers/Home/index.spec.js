import React from 'react'
import { shallow } from 'enzyme'
import { Home } from '../Home'

describe('Home container', () => {
    const props = {
        productsAll: {
            data: [{
                title: 'test1',
                products: []
            }],
            isLoading: false,
            errorMsg: null,
        },
        match: { params: { id: 1 }},
        fetchAllProducts: () => {}
    };

    const wrapper = shallow(<Home { ...props } />);
    it("render correctly Home container", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("render correctly product title prop", () => {
        expect(wrapper.find('Title').at(0).text()).toEqual(props.productsAll.data[0].title);
    });
});

