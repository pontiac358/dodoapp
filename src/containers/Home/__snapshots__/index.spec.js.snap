// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container render correctly Home container 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    fetchAllProducts={[Function]}
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    productsAll={
      Object {
        "data": Array [
          Object {
            "products": Array [],
            "title": "test1",
          },
        ],
        "errorMsg": null,
        "isLoading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <React.Fragment>
          <ForwardRef(render)>
            test1
          </ForwardRef(render)>
          <Connect(ProductList) />
        </React.Fragment>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(render)>
              test1
            </ForwardRef(render)>,
            <Connect(ProductList) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test1",
            },
            "ref": null,
            "rendered": "test1",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "h2",
              "__emotion_forwardProp": undefined,
              "__emotion_real": [Circular],
              "__emotion_styles": Array [
                "
    font-size: 26px;
    text-align:center;
    margin-bottom: 50px;  
",
              ],
              "defaultProps": undefined,
              "displayName": "Title",
              "render": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "products": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "__emotion_base": "div",
      "__emotion_forwardProp": undefined,
      "__emotion_real": [Circular],
      "__emotion_styles": Array [
        "
    max-width:1100px;
    margin:auto;   
",
      ],
      "defaultProps": undefined,
      "displayName": "Styled(div)",
      "render": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <React.Fragment>
            <ForwardRef(render)>
              test1
            </ForwardRef(render)>
            <Connect(ProductList) />
          </React.Fragment>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(render)>
                test1
              </ForwardRef(render)>,
              <Connect(ProductList) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "test1",
              },
              "ref": null,
              "rendered": "test1",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "h2",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "
    font-size: 26px;
    text-align:center;
    margin-bottom: 50px;  
",
                ],
                "defaultProps": undefined,
                "displayName": "Title",
                "render": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "products": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": undefined,
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          "
    max-width:1100px;
    margin:auto;   
",
        ],
        "defaultProps": undefined,
        "displayName": "Styled(div)",
        "render": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
